File: 206-ReverseLinkedList/206-ReverseLinkedList.py
# Last updated: 2/3/2026, 9:40:25 PM
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        
        prev=None
        curr=head

        while curr:
            temp=curr.next
            curr.next=prev
            prev=curr
            curr=temp
        return prev

        

File: 206-ReverseLinkedList/206-ReverseLinkedList.java
// Last updated: 2/3/2026, 9:40:36 PM
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {

        ListNode prev = null;
        ListNode curr = head;

        while (curr != null) {
            ListNode Temp = curr.next;
            curr.next = prev;
            prev=curr;
            curr=Temp; 
            
        }

        return prev;

    }
}